extra/onnx_ops.py

```diff
@@ -207,7 +207,7 @@ def Not(x:Tensor): return Where((x==1), Tensor.zeros(*x.shape), Tensor.ones(*x.s
 
 def ConstantOfShape(input, value=0.0):
   shape = [int(x) for x in safe_numpy(input)]
-  return Tensor.ones(*shape) * value
+  return Tensor.ones(*shape, dtype=value.dtype) * value if value > 0 and shape[0] > 0 else Tensor.zeros(*shape, dtype=value.dtype)
```

Comment:
Do you need this if statement?

Reply:
Without it test_constantofshape_int_shape_zero_cpu fails with AssertionError: zeros not allowed in shape (0,).
Tinygrad as far I understand doesn't support empty tensor operations with zero as torch does and one of the test cases test for a tensor of shape(0,) .

```python
torch.ones(0) * 0.0 # realizes into tensor([])
Tensor.ones(0) * 0.0 # AssertionError: zeros not allowed in shape (0,)
```

Maybe I'm missing something?

Reply:
Seems related to #777

Reply:
I Fetched Joqsan zero_dim_support branch and tried my changes there, but it still doesn't allow the behavior described above.

I could remove the if statement in my code and once the behavior is allowed in Tinygrad that test will pass automatically.